src/BlazorWebView/src/WindowsForms/BlazorWebView.cs

```diff
@@ -147,11 +148,22 @@ private void StartWebViewCoreIfPossible()
 
 			// We assume the host page is always in the root of the content directory, because it's
 			// unclear there's any other use case. We can add more options later if so.
-			var contentRootDir = Path.GetDirectoryName(Path.GetFullPath(HostPage));
-			var hostPageRelativePath = Path.GetRelativePath(contentRootDir, HostPage);
+			string appRootDir;
+			var entryAssemblyLocation = Assembly.GetEntryAssembly()?.Location;
+			if (!string.IsNullOrEmpty(entryAssemblyLocation))
```

Comment:
It might be handy to have a comment here about in what case `Assembly.GetEntryAssembly()?.Location` might not be set. Is this for something specific like Android where the assemblies are loaded in a special way?

Reply:
Oh actually this is WinForms so I guess not Android! Is it specifically with MSIX that the entry assembly has no location?

Reply:
Good point; I don't know of any case where it would be null in WinForms/WPF, so it's more of a "try to handle nulls gracefully" thing.

Reply:
I think Assembly.Location is null for single file exes. IIRC, something in the build (analyzer / trimmer / sdk) complains if you use that API.